saskatoon/unittests/test_permission.py

```diff
@@ -0,0 +1,44 @@
+import pytest
+from django.contrib.auth import get_user_model
+from pytest_django.asserts import assertRedirects
+from typing import Union
+
+STATIC_PAGES = ['calendar', 'property/create_public']
+STATIC_PAGES_PERMISSION_REQUIRED = ['harvest', 'property', 'organization', 'community', 'equipment', 'stats']
+
+User = get_user_model()
+
+
+def get_url(page_name: str) -> str:
+    return f"/{page_name}/"
+
+
+@pytest.mark.django_db
+def test_permission_static_pages_anonymous(client):
+    for page in STATIC_PAGES:
+        response = client.get(get_url(page))
+        assert response.status_code == 200, f"Page: {page}"
+
+    for page in STATIC_PAGES_PERMISSION_REQUIRED:
+        url = get_url(page)
+        response = client.get(url)
+        assertRedirects(response, f"/login/?next={url}", msg_prefix=f"Page: {page}")
+
+
+@pytest.mark.parametrize("auth_group, expected_status_codes",
+                         [('core', 200),
+                          ('pickleader', [200, 200, 200, 200, 200, 403]),
+                          ('volunteer', 403),
+                          ('owner', 403),
+                          ('contact', 403)])
+@pytest.mark.django_db
+def test_permission_static_pages(client, auth_group, expected_status_codes: Union[int, list]):
+    user = User.objects.create_user(email="test@user.com", password="password1234")
```

Comment:
I think it would be better to rely on our actual `AuthUser` model instead of not the built-in `User` and make use of the `add_role` or `set_roles utility functions:
https://github.com/LesFruitsDefendus/saskatoon-ng/blob/master/saskatoon/member/models.py#L81

Reply:
Here `User` is not the built-in one, and it is resolved to `AuthUser` at testing runtime (see line 9). I want to use `get_user_model()` to resolve it at runtime to shield the tests from implementation detail (plus, easier code refactoring in the future).

Reply:
Ok sorry I missed that, please rename it to `AuthUser` then.

Reply:
Done.